src/NuGet.Core/NuGet.Packaging/Signing/Authoring/SignPackageRequest.cs

```diff
@@ -37,14 +36,14 @@ public sealed class SignPackageRequest : IDisposable
         public X509Certificate2Collection AdditionalCertificates { get; }
 
         /// <summary>
-        /// Gets the repository V3 service index URL.
+        /// Gets the signature type.
         /// </summary>
-        public Uri V3ServiceIndexUrl { get; }
+        public SignatureType SignatureType { get; }
```

Comment:
This could be a virtual property overwritten in each subclass instead of storing it as a property, at the end of the day only the subclasses will be able to populate it.